pyvo/discover/tests/test_imagediscovery.py

```diff
@@ -0,0 +1,294 @@
+            time=time.Time("1970-10-13T13:00:00"))
+        d._query_one_sia2(queriable)
+
+        assert set(queriable.search_kwargs) == set(["time"])
+        assert abs(queriable.search_kwargs["time"][0].utc.value
+            - 40872.54166667) < 1e-8
+        assert abs(queriable.search_kwargs["time"][1].utc.value
+            - 40872.54166667) < 1e-8
+
+    def test_obscore(self):
+        queriable = FakeQueriable()
+        d = discover.ImageDiscoverer(
+            time=(time.Time("1970-10-13T13:00:00"), time.Time("1970-10-13T17:00:00")))
+        d.obscore_recs = [queriable]
+        d._query_obscore()
+
+        assert set(queriable.search_kwargs) == set()
+        assert queriable.search_args[0] == (
+            "select * from ivoa.obscore WHERE dataproduct_type='image'"
+            " AND (t_max>=40872.541666666664 AND 40872.708333333336>=t_min"
+            " AND t_min<=t_max AND 40872.541666666664<=40872.708333333336)")
+
+    def test_obscore_nointerval(self):
+        queriable = FakeQueriable()
+        d = discover.ImageDiscoverer(
+            time=(
+                time.Time("1970-10-13T13:00:00")))
+        d.obscore_recs = [queriable]
+        d._query_obscore()
+
+        assert set(queriable.search_kwargs) == set()
+        assert queriable.search_args[0] == (
+            "select * from ivoa.obscore WHERE dataproduct_type='image'"
+            " AND (t_max>=40872.541666666664 AND 40872.541666666664>=t_min"
+            " AND t_min<=t_max AND 40872.541666666664<=40872.541666666664)")
+
+
+class TestSpaceCondition:
+    def test_sia1_fails_without_space(self):
+        queriable = FakeQueriable()
+        di = discover.ImageDiscoverer(
+            time=(
+                time.Time("1970-10-13T13:00:00")))
+        di.sia1_recs = [queriable]
+        di.query_services()
+        assert di.log_messages == [
+            'SIA1 service(s) skipped due to missing space constraint']
+
+    def test_sia2(self):
+        queriable = FakeQueriable()
+        di = discover.ImageDiscoverer(space=(30, 21, 1))
+        di.sia2_recs = [queriable]
+        di.query_services()
+        assert queriable.search_kwargs == {'pos': (30, 21, 1)}
+
+    def test_obscore(self):
+        queriable = FakeQueriable()
+        di = discover.ImageDiscoverer(space=(30, 21, 1))
+        di.obscore_recs = [queriable]
+        di.query_services()
+        assert queriable.search_args == (
+            "select * from ivoa.obscore WHERE dataproduct_type='image'"
+            " AND (1=contains(point('ICRS', s_ra, s_dec),"
+            " circle('ICRS', 30, 21, 1))"
+            " or 1=intersects(circle(30, 21, 1), s_region))",)
+
+
+def test_no_services_selected():
+    with pytest.raises(dal.DALQueryError) as excinfo:
+        image.ImageDiscoverer().query_services()
+    assert "No services to query." in str(excinfo.value)
+
+
+# Tests requiring remote data below this line
+
+
+@pytest.mark.remote_data
+def test_single_sia1():
```

Comment:
This test is failing for me with 

```
E           pyvo.dal.exceptions.DALQueryError: No services to query.  Unless you overrode service selection, you will have to loosen your constraints.
```

Reply:
Ahem.  I accidentally unpublished that service as version 2 of that resource came in.  That particular service will remain, though, and I've re-published it.  The test succeeds again.